module/Application/src/Controller/StatusController.php

```diff
@@ -122,59 +161,53 @@ public function get($ids)
 
         $exploded_ids = explode(',', $ids);
         $results = [];
+        $idsToCheckInSirius = [];
 
         foreach ($exploded_ids as $id) {
-            $this->getLogger()->debug('Checking Sirius status for ' . $id);
-
-            $lpaResult = $this->getService()->fetch($id, $this->routeUserId);
-
-            // if the id isn't found, return false.
-            if ($lpaResult instanceof ApiProblem) {
-                $this->getLogger()->err('Error accessing LPA data: ' . $lpaResult->getDetail());
+            $currentProcessingStatus = $this->getCurrentProcessingStatus($id);
 
+            if ($currentProcessingStatus instanceof ApiProblem) {
                 $results[$id] = ['found' => false];
                 continue;
             }
 
-            /** @var Lpa $lpa */
-            $lpa = $lpaResult->getData();
-
-            $metaData = $lpa->getMetaData();
-
-            $currentProcessingStatus = array_key_exists(LPA::SIRIUS_PROCESSING_STATUS, $metaData) ?
-                $metaData[LPA::SIRIUS_PROCESSING_STATUS] : null;
+            if ($currentProcessingStatus == null) {
+                $results[$id] = ['found' => false];
+            } else {
+                $results[$id] = ['found' => true, 'status' => $currentProcessingStatus];
+            }
 
-            // If application has already reached the last stage of processing do not check for updates
-            if ($currentProcessingStatus == Lpa::SIRIUS_PROCESSING_STATUS_RETURNED) {
-                $results[$id] = ['found' => true, 'status' => Lpa::SIRIUS_PROCESSING_STATUS_RETURNED];
-                continue;
+            //Add id's to array, to check updates in Sirius for applications that has not reached the last stage of processing.
+            if ($currentProcessingStatus != Lpa::SIRIUS_PROCESSING_STATUS_RETURNED) {
+                $idsToCheckInSirius[] = $id;
             }
+        }
+        // Get status update from Sirius
+        if ($idsToCheckInSirius != null) {
```

Comment:
I don't think would ever be null. Maybe `!empty()`?

Reply:
done